build/parseSpec.c

```diff
@@ -462,6 +430,22 @@ int readLine(rpmSpec spec, int strip)
     lineType = parseLineType(s);
     if (!lineType)
 	goto after_classification;
+
+    /* check ordering of the conditional */
+    if (lineType->isConditional &&
+	(spec->readStack->lastConditional->id & lineType->wrongPrecursors)) {
```

Comment:
The condition continuation is on the same indentation level as the if-block it surrounds, which makes it easy to misread. Please fix the indentation, using well named helper variables to shorten the condition is usually the best way to do that.

This was probably there in previous rounds too, sorry for not noticing.